### PR TITLE
Resize find/replace filters to 1024 bytes

### LINKED ISSUE #12041: [bug] "Find in Files": Long Filters Get Truncated
### Description of the Issue
"Find in Files": When using a long filter, then the list gets truncated and not all patterns are included/excluded.

### Steps to Reproduce the Issue

1.  Use "Find in Files" with following filter `* !+\.svn !+\.git !*.exe !*.com !*.zip !*.gz !*.7z !*.cab !*.deb !*.pkg !*.bmp !*.jpg !*.jpeg !*.png !*.gif !*.ico !*.crt !*.pub !*.pem !*.key !*.csr !*.pdf !*.docx !*.doc !*.odt !*.xlsx !*.xls !*.ods !*.mdb !*.epub !*.drawio !*.swf !*.jar !*.wmv !*.wav !*.mp4 !*.mp3 !*.iso !*.cso !*.vpk !*.cmbackup !*.bin !+\dvb !+\cache2` (324 chars)
2. Execute the search

### Expected Behavior
The filter is untouched and complete

### Actual Behavior
It is cut down to 255chars: `* !+\.svn !+\.git !*.exe !*.com !*.zip !*.gz !*.7z !*.cab !*.deb !*.pkg !*.bmp !*.jpg !*.jpeg !*.png !*.gif !*.ico !*.crt !*.pub !*.pem !*.key !*.csr !*.pdf !*.docx !*.doc !*.odt !*.xlsx !*.xls !*.ods !*.mdb !*.epub !*.drawio !*.swf !*.jar !*.wmv !*.wav !`
Please increase to 1024 (1k).

### Debug Information
Notepad++ v8.4.4   (32-bit)
Build time : Jul 15 2022 - 17:56:38
Path : C:\Program Files (x86)\Notepad++\notepad++.exe
Command Line : "D:\Users\private\Documents\Coding\regex.txt"
Admin mode : OFF
Local Conf mode : OFF
Cloud Config : OFF
OS Name : Windows 10 Enterprise (64-bit) 
OS Version : 21H2
OS Build : 19044.1889
Current ANSI codepage : 1252
Plugins : 
    ComparePlugin (2.0.2)
    CSVLint (0.4.5.2)
    ElasticTabstops (1.3.1)
    HexEditor (0.9.12)
    JSMinNPP (1.2205)
    mimeTools (2.8)
    NppConverter (4.4)
    NppExport (0.4)
    RegexTrainer (1)
    XMLTools (3.1.1.13)

### PR DESCRIPTION
**Pull request by doug1234**
Fix #12041

### CODE REVIEW

File: PowerEditor/src/MISC/Common/Common.cpp [hunk 1]
```diff
@@ -1137,6 +1137,20 @@ HWND CreateToolTipRect(int toolID, HWND hWnd, HINSTANCE hInst, const PTSTR pszTe
 	return hwndTip;
 }
 
+generic_string GetDialogItemString(HWND dialog, int item)
+{
+	constexpr int startSize = 256;
+	generic_string res(startSize, '\0');
+	auto stringSize = ::GetDlgItemText(dialog, item, &res[0], res.size());
```
Comment (rdipardo) [inline PowerEditor/src/MISC/Common/Common.cpp:1144]: `res.size()` needs an explicit cast.
`std::wstring::size()` returns a `size_t`; on x86_64 machines, that's equal to `unsigned __int64`.

https://ci.appveyor.com/project/donho/notepad-plus-plus/builds/44667854/job/x8e9badc7uh0m344/messages

Comment (doug1234) [reply PowerEditor/src/MISC/Common/Common.cpp]: Sorry about that.  Should be fixed now.

**Comment by alankilborn** [comment]
Why did this start with fixes for #11988 and then jump over to issue #12041 ???

**Comment by doug1234** [comment]
> Why did this start with fixes for #11988 and then jump over to issue #12041 ???

All the fixes here are for #12041.  Not sure why the confusion.

**Comment by CennoxX** [comment]
> > Why did this start with fixes for #11988 and then jump over to issue #12041 ???
> 
> All the fixes here are for #12041. Not sure why the confusion.

Your first few commits of the PR all reference #11988:
grafik

File: PowerEditor/src/ScintillaComponent/FindReplaceDlg.cpp [hunk 2]
```diff
@@ -1685,13 +1685,9 @@ intptr_t CALLBACK FindReplaceDlg::run_dlgProc(UINT message, WPARAM wParam, LPARA
 				case IDD_FINDINFILES_FIND_BUTTON:
 				{
 					setStatusbarMessage(TEXT(""), FSNoMessage);
-					const int filterSize = 256;
```
Comment (donho) [inline PowerEditor/src/ScintillaComponent/FindReplaceDlg.cpp:1688]: Modifying the value from 256 to 1024 is enough fairly for the filter size.

**Review by donho — CHANGES_REQUESTED** [review]
Please follow the given suggestion

File: PowerEditor/src/ScintillaComponent/FindReplaceDlg.cpp (hunk 2 above)
Comment (doug1234) [reply PowerEditor/src/ScintillaComponent/FindReplaceDlg.cpp:1688]: OK.  I can do that tomorrow if you are sure you don't want a dynamic size.  I think there are probably 3 places it should be changed.

Comment (alankilborn) [reply PowerEditor/src/ScintillaComponent/FindReplaceDlg.cpp:1688]: And you really need to get this out of "combination" with 11988!

Comment (donho) [reply PowerEditor/src/ScintillaComponent/FindReplaceDlg.cpp:1688]: @alankilborn 
Sorry I don't follow you. Could you elaborate it?

Comment (donho) [reply PowerEditor/src/ScintillaComponent/FindReplaceDlg.cpp:1688]: @alankilborn OK, I see.
Yes, it's confusing. It's due to the wrong number in the previous commit.

Comment (doug1234) [reply PowerEditor/src/ScintillaComponent/FindReplaceDlg.cpp:1688]: So just to clarify, remove the function that reads a variable size string and just bump the buffer size to 1024?

Comment (donho) [reply PowerEditor/src/ScintillaComponent/FindReplaceDlg.cpp:1688]: Yes. Just modify `const int filterSize = 256;` to const int filterSize = 1024; from the initial state of code source.
The added function is not necessary.

Comment (doug1234) [reply PowerEditor/src/ScintillaComponent/FindReplaceDlg.cpp:1688]: @donho It should match what you suggested now.

**Review by donho — APPROVED** [review]

**Comment by donho** [comment]
@doug1234
During rebasing your PR, I've got the conflict, shown following:
 
```
$ git rebase -i
error: could not apply d9b37771... Added the ability to turn off selecting text when displaying the Field dialog for issue #11988.
Resolve all conflicts manually, mark them as resolved with
"git add/rm <conflicted_files>", then run "git rebase --continue".
You can instead skip this commit: run "git rebase --skip".
To abort and get back to the state before "git rebase", run "git rebase --abort".
Could not apply d9b37771... Added the ability to turn off selecting text when displaying the Field dialog for issue #11988.
Auto-merging PowerEditor/src/WinControls/Preference/preference_rc.h
CONFLICT (content): Merge conflict in PowerEditor/src/WinControls/Preference/preference_rc.h
Auto-merging PowerEditor/src/WinControls/Preference/preferenceDlg.cpp
CONFLICT (content): Merge conflict in PowerEditor/src/WinControls/Preference/preferenceDlg.cpp
Auto-merging PowerEditor/src/WinControls/Preference/preference.rc
CONFLICT (content): Merge conflict in PowerEditor/src/WinControls/Preference/preference.rc
Auto-merging PowerEditor/src/Parameters.h
CONFLICT (content): Merge conflict in PowerEditor/src/Parameters.h
Auto-merging PowerEditor/src/Parameters.cpp
CONFLICT (content): Merge conflict in PowerEditor/src/Parameters.cpp
Auto-merging PowerEditor/src/NppCommands.cpp
CONFLICT (content): Merge conflict in PowerEditor/src/NppCommands.cpp
Auto-merging PowerEditor/src/NppBigSwitch.cpp
CONFLICT (content): Merge conflict in PowerEditor/src/NppBigSwitch.cpp
```

Therefore this PR won't be merged. Please redo a PR based the current state of master.

**Pull request closed without merging**